### DISCUSSION TITLE
Importing functions into Cython with cimport

### DISCUSSION THREAD

**Discussion by YevgeniySimonov** [rapidfuzz/RapidFuzz#202] [Q&A] [answered]
Good day! I am using your library in one of my projects that requires high computational speed and it is performing very well. However, I would like to parallelise some nested loops performing numerous calls to partial_ratio. I have translated my code into cython and wondering how I can import functions, such as rapidfuzz.fuzz.partial_ratio into my cython project with cimport?
Regards,
Yevgeniy

**Comment by maxbachmann** [comment, marked as answer]
There are a couple of ways this can be done:
1) maybe the process functions already do what you need
    -> call e.g. `process.cdist` instead of calling `partial_ratio` multiple times
2) The python project is just a wrapper for the C++ library. So you can directly use https://github.com/maxbachmann/rapidfuzz-cpp inside your application in a similar way to this wrapper.
3) RapidFuzz provides a C-API, which is just not documented yet. Every scorer in rapidfuzz provides a PyCapsule which allows you to directly call scorers. This API is e.g. used inside the process module (see https://github.com/maxbachmann/RapidFuzz/blob/main/rapidfuzz/cpp_process.pyx). Large parts of this C-API can be used without the GIL, so it allows multithreading (I do this in process.cdist). I should be able to write a small documentation on the usage of the C-API this weekend.
